pkg/meta/version.go

```diff
@@ -2,7 +2,7 @@ package meta
 
 const (
 	// CLIAPIVersion used to communicate with user e.g. longhorn-manager
-	CLIAPIVersion    = 8
+	CLIAPIVersion    = 9
 	CLIAPIMinVersion = 3
 
 	// ControllerAPIVersion used to communicate with instance-manager
```

Comment:
I think you are right. It seems CLIAPIVersion relates to whether or not I have changed CLI flags and ControllerAPIVersion relates to whether or not I have changed gRPC fields? Since I have done both, both should be bumped? And since in all cases I have made the new flags/fields optional, I do not need to bump MinVersions?

I too am confused on exactly how ControllerAPIVersion is used. instance-manager does not reference it, but maybe we are thinking this way:

If there is already an engine successfully running in an instance-manager with ControllerAPIVersion = x and MinControllerAPIVersion = y, clearly instance-manager is able to handle those values (even though instance-manager itself is not checking them). As long as we don't decrease ControllerAPIVersion (say to x - 1) or increase MinControllerAPIVersion (say to y + 1), the same instance-manager can run the new engine?

Reply:
> It seems CLIAPIVersion relates to whether or not I have changed CLI flags and ControllerAPIVersion relates to whether or not I have changed gRPC fields? Since I have done both, both should be bumped? And since in all cases I have made the new flags/fields optional, I do not need to bump MinVersions?

This is what I expected as well. But got confused about how ControllerAPIVersion is used as discussed above.

> If there is already an engine successfully running in an instance-manager with ControllerAPIVersion = x and MinControllerAPIVersion = y, clearly instance-manager is able to handle those values (even though instance-manager itself is not checking them). As long as we don't decrease ControllerAPIVersion (say to x - 1) or increase MinControllerAPIVersion (say to y + 1), the same instance-manager can run the new engine?

But in practice, we did increase MinControllerAPIVersion several times. So I am expecting there is some check in instance-manager but couldn't find any. Maybe this field is not used at all and we should combine it into CLIAPIVersion?